### PR TITLE
JIT insert of shader dictionary

### PR DESCRIPTION
**Pull request by jayhover**

### CODE REVIEW

**Comment by jayhover** [comment]
replace the hardcoded dj2?

**Comment by elfrank** [comment]
@jayhover how to run build?
`npm run build-uglify`

Reference: https://github.com/mrdoob/three.js/wiki/Build-instructions

File: build/three.js [hunk 1]
```diff
@@ -16490,6 +16490,15 @@
 
 	}
 
+	function djb2Code(str){
+	    var hash = 5381;
+	    for (i = 0; i < str.length; i++) {
```
Comment (elfrank) [inline build/three.js]: @jayhover `i` is not defined

**Review by elfrank — CHANGES_REQUESTED** [review]

**Comment by elfrank** [comment]
@jayhover after your fix, remember to run `npm run build-uglify` and push the build changes as well.

File: build/three.js (hunk 1 above)
Comment (jayhover) [reply build/three.js]: err I thought I fixed that... hmm

**Comment by jayhover** [comment]
pushed ( uglified )

File: build/three.js [hunk 2]
```diff
@@ -16490,6 +16490,15 @@
 
 	}
 
+	function djb2Code(str){
+	    var hash = 5381;
+	    for (var i = 0; i < str.length; i++) {
+	        char = str.charCodeAt(i);
```
Comment (elfrank) [inline build/three.js]: @jayhover `char` not defined

**Review by elfrank — CHANGES_REQUESTED** [review]

**Comment by jayhover** [comment]
just pushed a few more flags to provide mode perf info.

File: src/renderers/webgl/WebGLProgram.js [hunk 3]
```diff
@@ -194,6 +194,15 @@ function unrollLoops( string ) {
 
 }
 
+function djb2Code(str){
+    var hash = 5381;
+    for (var i = 0; i < str.length; i++) {
+        char = str.charCodeAt(i);
```
Comment (elfrank) [inline src/renderers/webgl/WebGLProgram.js]: @jayhover `char` is not defined

**Review by elfrank — CHANGES_REQUESTED** [review]

**Pull request closed without merging**